splinepy/helpme/create.py

```diff
@@ -277,6 +284,131 @@ def from_bounds(parametric_bounds, physical_bounds):
     return bspline_box
 
 
+def determinant_spline(self):
+    """
+    Creates a spline representing the projection of the jacobian determinant
+
+    Only works if (parameter dimension == physical dimension) and non-rational splines
+    or rational splines with equal weights
+    (otherwise give B-Spline approximation of spline)!
+
+    dmin > 0 --> spline is not tangled
+    dmin <= 0 --> spline is COULD BE tangled
+
+    references:
+    Mantzaflaris, A., Jüttler, B., Khoromskij, B. N., & Langer, U. (2017)
+    Limkilde, A., Evgrafov, A., Gravesen, J., & Mantzaflaris, A. (2021)
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    determinant_projection: BSpline
+      Spline which represents the jacobian determinant of the given spline object.
+    """
+
+    from splinepy import Bezier, BSpline
+
+    # checks if dimensions are equal.
+    if self.dim != self.para_dim:
+        raise ValueError(
+            f"Dimension of parameter space and physical space ({self.whatami})"
+            "are not the same!"
+        )
+
+    # checks if spline is rational and if weights differ
+    # otherwise NURBS can be treated like regular BSpline
+
+    if self.is_rational and np.allclose(self.weights, self.weights.ravel()[0]):
+        rational_spline = True
+        log.debug(
+            f"Spline is rational ({self.whatami}) with unequal weights!"
+            "Determinant Spline will be calculated with all weights set to 1."
+        )
+        # save old weights
+        old_weights = self.weights
+        # overwrite weights of spline
+        self.weights = np.repeat([1], len(old_weights))
+    else:
+        rational_spline = False
+
+    # calculate degrees of det spline
+    deg_det = self.degrees * self.dim - 1
+    d = self.para_dim
+    # initialize lists for knot vectors
+    kvs_det = [None] * d
+    # calculate necessary additional knot multiplicity due to degree elevation
+    mult_inc = self.degrees * (d - 1) - 1
+
+    # knot_vectors
+    if self.has_knot_vectors:
+        for i in range(d):
+            # Add knots ([inner knots] * (mult_inc + 1))
+            # since continuity at inner knots further decreases by 1!
+            kvs_temp = np.array(self.knot_vectors[i]).tolist()
+            unique_knots = np.unique(kvs_temp).tolist()
+            kvs_temp += np.repeat(unique_knots[1:-1], mult_inc[i] + 1).tolist()
+            kvs_temp += np.repeat(
+                [unique_knots[0], unique_knots[-1]], mult_inc[i]
+            ).tolist()
+            kvs_temp.sort()
+            kvs_det[i] = kvs_temp
+
+        # number of cpts
+        n_cpts = np.prod([len(kvs_det[i]) - deg_det[i] - 1 for i in range(d)])
+
+        # create spline with empty cpts for calculation
+        determinant_projection = BSpline(
+            degrees=deg_det,
+            control_points=np.empty((n_cpts, 1)),
+            knot_vectors=kvs_det,
+        )
```

Comment:
Can we have some function like `empty_spline(dim, degrees, kvs)` that does exactly this? This doesn't need to be this PR.

Reply:
@jzwar: Next PR!